test/src/test/java/jenkins/security/AgentSecretActionTest.java

```diff
@@ -0,0 +1,108 @@
+    private StaplerResponse2 rsp;
+    private StringWriter stringWriter;
+    private PrintWriter writer;
+
+    @Before
+    public void setUp() throws Exception {
+        SlaveComputer computer = mock(SlaveComputer.class);
+        action = new AgentSecretAction(computer);
+        req = mock(StaplerRequest2.class);
+        rsp = mock(StaplerResponse2.class);
+        stringWriter = new StringWriter();
+        writer = new PrintWriter(stringWriter);
+        when(rsp.getWriter()).thenReturn(writer);
+    }
+
+    @Test
+    public void testGetSecretWithValidPermissions() throws Exception {
+        DumbSlave agent = j.createSlave("test-agent", null);
+        String expectedSecret = ((SlaveComputer) agent.getComputer()).getJnlpMac();
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Computer.CONNECT).everywhere().to("test-user"));
+
+        action.doGet(req, rsp, "test-agent");
+        writer.flush();
+        verify(rsp).setContentType("text/plain");
+        assertEquals(expectedSecret, stringWriter.toString());
+    }
+
+    @Test
+    public void testGetSecretWithoutPermissions() throws Exception {
+        j.createSlave("test-agent", null);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy());
+
+        thrown.expect(org.acegisecurity.AccessDeniedException.class);
+        action.doGet(req, rsp, "test-agent");
+    }
+
+    @Test
+    public void testGetSecretWithInvalidNodeName() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Computer.CONNECT).everywhere().to("test-user"));
+
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Node not found: non-existent-agent");
+        action.doGet(req, rsp, "non-existent-agent");
+    }
+
+    @Test
+    public void testGetSecretWithNullNodeName() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Computer.CONNECT).everywhere().to("test-user"));
+
+
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Node name is required");
+        action.doGet(req, rsp, null);
+    }
+
+    @Test
+    public void testGetSecretWithMasterNode() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Computer.CONNECT).everywhere().to("test-user"));
+
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("The specified node is not an agent/slave node: master");
+        action.doGet(req, rsp, "master");
```

Comment:
The built-in node has no name that you can ask like that. Also the term `master` is not used since years. With the action being bound to SlaveComputer you will never be able to access it on the built-in node.
Suggesting to completely remove this method